insights/core/spec_factory.py

```diff
@@ -126,10 +127,9 @@ def content(self):
 
     def write(self, dst):
         fs.ensure_path(os.path.dirname(dst))
-        # Clean Spec Content when writing it down to disk and before uploading
-        self._clean_content()
+        # Clean Spec Content when writing it down to disk before uploading
         with open(dst, "wb") as f:
-            content = "\n".join(self.content)
+            content = "\n".join(self._clean_content())
```

Comment:
@xiangce What about keeping this `self._clean_content()` calling before the file open phrase? When `ContentException` be raised during `self._clean_content()`, it can save a file open action somehow.

Reply:
Nice catching @JoySnow. It would also prevent creating an empty file. updated in the new commit.